src/TriggerBinding/SqlTriggerListener.cs

```diff
@@ -0,0 +1,366 @@
+                this._logger.LogInformation($"Stopped SQL trigger listener for table: '{this._userTable.FullName}', function ID: '{this._userFunctionId}'.");
+            }
+
+            return Task.CompletedTask;
+        }
+
+        /// <summary>
+        /// Returns the object ID of the user table.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">Thrown in case of error when querying the object ID for the user table</exception>
+        private async Task<int> GetUserTableIdAsync(SqlConnection connection, CancellationToken cancellationToken)
+        {
+            string getObjectIdQuery = $"SELECT OBJECT_ID(N{this._userTable.QuotedFullName}, 'U');";
+
+            using (var getObjectIdCommand = new SqlCommand(getObjectIdQuery, connection))
+            using (SqlDataReader reader = await getObjectIdCommand.ExecuteReaderAsync(cancellationToken))
+            {
+                if (!await reader.ReadAsync(cancellationToken))
+                {
+                    throw new InvalidOperationException($"Received empty response when querying the object ID for table: '{this._userTable.FullName}'.");
+                }
+
+                object userTableId = reader.GetValue(0);
+
+                if (userTableId is DBNull)
+                {
+                    throw new InvalidOperationException($"Could not find table: '{this._userTable.FullName}'.");
+                }
+
+                return (int)userTableId;
+            }
+        }
+
+        /// <summary>
+        /// Gets the names and types of primary key columns of the user table.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">
+        /// Thrown if there are no primary key columns present in the user table or if their names conflict with columns in worker table.
+        /// </exception>
+        private async Task<IReadOnlyList<(string name, string type)>> GetPrimaryKeyColumnsAsync(SqlConnection connection, int userTableId, CancellationToken cancellationToken)
+        {
+            string getPrimaryKeyColumnsQuery = $@"
+                SELECT c.name, t.name, c.max_length, c.precision, c.scale
+                FROM sys.indexes AS i
+                INNER JOIN sys.index_columns AS ic ON i.object_id = ic.object_id AND i.index_id = ic.index_id
+                INNER JOIN sys.columns AS c ON ic.object_id = c.object_id AND ic.column_id = c.column_id
+                INNER JOIN sys.types AS t ON c.user_type_id = t.user_type_id
+                WHERE i.is_primary_key = 1 AND i.object_id = {userTableId};
+            ";
+
+            using (var getPrimaryKeyColumnsCommand = new SqlCommand(getPrimaryKeyColumnsQuery, connection))
+            using (SqlDataReader reader = await getPrimaryKeyColumnsCommand.ExecuteReaderAsync(cancellationToken))
+            {
+                string[] reservedColumnNames = new string[] { "ChangeVersion", "AttemptCount", "LeaseExpirationTime" };
```

Comment:
I know you said you didn't use constants because of code readability, but I don't like the potential risk of regression here by having these values be separate. So I'd strongly prefer that we make constants for each of these - even if it does decrease the readability of the query slightly.

Reply:
Just to make sure there is no disconnect, I will be updating all SQL queries to use constants for column names in worker table. That was what I mentioned in the other conversation.

Reply:
This one requires discussion before I push any change. I have added it to #282 to ensure this gets discussed and work gets done.